### ISSUE TITLE
Installation issue

### ISSUE THREAD

**Issue by idc9** [pkimes/sigclust2#7]
When I install the package using the provided instructions I encounter an error.

```R
> devtools::install_github("pkimes/sigclust2")
Downloading GitHub repo pkimes/sigclust2@master
from URL https://api.github.com/repos/pkimes/sigclust2/zipball/master
Installing sigclust2
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/dplyr_0.7.7.tgz'
Content type 'application/x-gzip' length 5198161 bytes (5.0 MB)
==================================================
downloaded 5.0 MB

Installing dplyr
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/glue_1.3.0.tgz'
Content type 'application/x-gzip' length 93367 bytes (91 KB)
==================================================
downloaded 91 KB

Installing glue
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a6a61d1ea/glue'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘glue’ ...
* DONE (glue)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/pkgconfig_2.0.2.tgz'
Content type 'application/x-gzip' length 15456 bytes (15 KB)
==================================================
downloaded 15 KB

Installing pkgconfig
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a49c2c4c/pkgconfig'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘pkgconfig’ ...
* DONE (pkgconfig)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/R6_2.3.0.tgz'
Content type 'application/x-gzip' length 34492 bytes (33 KB)
==================================================
downloaded 33 KB

Installing R6
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a3f45b26/R6'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘R6’ ...
* DONE (R6)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/Rcpp_0.12.19.tgz'
Content type 'application/x-gzip' length 4429794 bytes (4.2 MB)
==================================================
downloaded 4.2 MB

Installing Rcpp
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a7285bc43/Rcpp'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘Rcpp’ ...
* DONE (Rcpp)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/rlang_0.3.0.1.tgz'
Content type 'application/x-gzip' length 1074663 bytes (1.0 MB)
==================================================
downloaded 1.0 MB

Installing rlang
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a78473bd6/rlang'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘rlang’ ...
* DONE (rlang)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/tidyselect_0.2.5.tgz'
Content type 'application/x-gzip' length 294992 bytes (288 KB)
==================================================
downloaded 288 KB

Installing tidyselect
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a3171858a/tidyselect'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘tidyselect’ ...
* DONE (tidyselect)
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a1344b4cb/dplyr'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘dplyr’ ...
* DONE (dplyr)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/ggdendro_0.1-20.tgz'
Content type 'application/x-gzip' length 132139 bytes (129 KB)
==================================================
downloaded 129 KB

Installing ggdendro
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/ggplot2_3.1.0.tgz'
Content type 'application/x-gzip' length 3179264 bytes (3.0 MB)
==================================================
downloaded 3.0 MB

Installing ggplot2
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/digest_0.6.18.tgz'
Content type 'application/x-gzip' length 166752 bytes (162 KB)
==================================================
downloaded 162 KB

Installing digest
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a7ce7e9c/digest'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘digest’ ...
* DONE (digest)
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/scales_1.0.0.tgz'
Content type 'application/x-gzip' length 590050 bytes (576 KB)
==================================================
downloaded 576 KB

Installing scales
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a628434f4/scales'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘scales’ ...
* DONE (scales)
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a107752db/ggplot2'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘ggplot2’ ...
* DONE (ggplot2)
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a9f50133/ggdendro'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘ggdendro’ ...
* DONE (ggdendro)
Skipping install of 'ggplot2' from a cran remote, the SHA1 (3.1.0) has not changed since last install.
  Use `force = TRUE` to force installation
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/ggthemes_4.0.1.tgz'
Content type 'application/x-gzip' length 334081 bytes (326 KB)
==================================================
downloaded 326 KB

Installing ggthemes
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a7a815897/ggthemes'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

* installing *binary* package ‘ggthemes’ ...
* DONE (ggthemes)
Skipping install of 'Rcpp' from a cran remote, the SHA1 (0.12.19) has not changed since last install.
  Use `force = TRUE` to force installation
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/tf/8lz5km5n0n509s9y_f1wpj3h0000gn/T/RtmpNU8h6m/devtools1564a4a3a386/pkimes-sigclust2-f5b57f0'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests

ERROR: dependencies ‘Rclusterpp’, ‘WGCNA’ are not available for package ‘sigclust2’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/sigclust2’
Installation failed: Command failed (1)

```

My R/computer setup is as follows
```R
> R.Version()
$platform
[1] "x86_64-apple-darwin15.6.0"

$arch
[1] "x86_64"

$os
[1] "darwin15.6.0"

$system
[1] "x86_64, darwin15.6.0"

$status
[1] ""

$major
[1] "3"

$minor
[1] "4.3"

$year
[1] "2017"

$month
[1] "11"

$day
[1] "30"

$`svn rev`
[1] "73796"

$language
[1] "R"

$version.string
[1] "R version 3.4.3 (2017-11-30)"

$nickname
[1] "Kite-Eating Tree"


```

**Comment by idc9** [comment]
I updated my R version and get a similar error

```R
> devtools::install_github("pkimes/sigclust2")
Downloading GitHub repo pkimes/sigclust2@master
Skipping 5 packages not available: AnnotationDbi, GO.db, impute, preprocessCore, Rclusterpp
Installing 62 packages: acepack, AnnotationDbi, BH, bindr, bindrcpp, checkmate, colorspace, data.table, DEoptimR, doParallel, dplyr, dynamicTreeCut, evaluate, fansi, fastcluster, fit.models, foreach, Formula, ggdendro, ggplot2, ggthemes, GO.db, gridExtra, gtable, highr, Hmisc, htmlTable, htmltools, htmlwidgets, impute, iterators, knitr, labeling, latticeExtra, lazyeval, markdown, matrixStats, munsell, mvtnorm, pcaPP, pillar, pkgconfig, plogr, plyr, preprocessCore, purrr, Rclusterpp, RColorBrewer, reshape2, robust, robustbase, rrcov, scales, stringi, stringr, tibble, tidyselect, utf8, viridis, viridisLite, WGCNA, yaml
Error: (converted from warning) packages ‘AnnotationDbi’, ‘GO.db’, ‘impute’, ‘preprocessCore’, ‘Rclusterpp’ are not available (for R version 3.5.1)
```

```R

> R.Version()
$platform
[1] "x86_64-apple-darwin15.6.0"

$arch
[1] "x86_64"

$os
[1] "darwin15.6.0"

$system
[1] "x86_64, darwin15.6.0"

$status
[1] ""

$major
[1] "3"

$minor
[1] "5.1"

$year
[1] "2018"

$month
[1] "07"

$day
[1] "02"

$`svn rev`
[1] "74947"

$language
[1] "R"

$version.string
[1] "R version 3.5.1 (2018-07-02)"

$nickname
[1] "Feather Spray"

```

**Comment by pkimes** [comment]
Hi @idc9 - just got a chance to look at this. I think root of your problem is:

1. `Rclusterpp` is [no longer available on CRAN](https://cran.r-project.org/web/packages/Rclusterpp/index.html) and needs to be installed from [this GitHub repo].(https://github.com/nolanlab/Rclusterpp)
2. `WGCNA` depends on some packages on Bioconductor (`‘AnnotationDbi’, ‘GO.db’, ‘impute’, ‘preprocessCore’`) and not CRAN.

I'll add a note to the README about `Rclusterpp` for now, and try to update the package to remove this deprecated dependency when I get a chance. 

Thanks for reporting this. 
Let me know if the above solves your problems!

**Comment by pkimes** [comment]
I've removed the dependence on `Rclusterpp` and moved this to "`Enhances:`" in the DESCRIPTION file in commit 55288e5. The package is no longer required, but if installed, can still be used with `sigclust2`. The package passed R CMD build/check on machines with and without `Rclusterpp` installed, so I'm going to close this issue. 

Let me know if you still run into issues.

**Comment by idc9** [comment]
I'm getting a different error now

```R
> devtools::install_github("pkimes/sigclust2")
Downloading GitHub repo pkimes/sigclust2@master
These packages have more recent versions available.
Which would you like to update?

1:   dplyr (0.7.7 -> 0.7.8) [CRAN]

Enter one or more numbers separated by spaces, or an empty line to cancel
1: 
Installing 26 packages: acepack, checkmate, data.table, DEoptimR, doParallel, dynamicTreeCut, fastcluster, fit.models, foreach, Formula, ggdendro, ggthemes, gridExtra, Hmisc, htmlTable, htmlwidgets, iterators, latticeExtra, matrixStats, mvtnorm, pcaPP, robust, robustbase, rrcov, viridis, WGCNA
Error: (converted from warning) dependencies ‘impute’, ‘preprocessCore’, ‘GO.db’, ‘AnnotationDbi’ are not available
```

**Comment by SamGG** [comment]
‘impute’, ‘preprocessCore’, ‘GO.db’, ‘AnnotationDbi’ are Bioconductor packages. I think you should install them first.

**Comment by pkimes** [comment]
Exactly what @SamGG said. (Thanks @SamGG !)

@idc9, Bioconductor is like CRAN, but contains genomics/bioinformatics packages:
http://bioconductor.org/install/

**Comment by idc9** [comment]
Ah good point, I should have thought about that. You might add a message to the README e.g.

Some packages need to be installed from [Bioconductor](https://www.bioconductor.org/install/)
```R
BiocManager::install(c("impute", "preprocessCore", "GO.db", "AnnotationDbi"))
```

**Comment by SamGG** [comment]
BTW, one can use BiocManager::install() for Bioconductor's packages but also CRAN's ones and github repos (without devtools).
One should try `BiocManager::install("pkimes/sigclust2")` and maybe everything will be installed smoothly without the need of using different commands depending on the repository.

**Comment by pkimes** [comment]
Thanks again, @SamGG!

@idc9 - I've updated the README to hopefully make installation easier for users. As pointed out by @SamGG, it looks like installing `sigclust2` using a single call to `BiocManager::install` should handle everything without requiring the user to install the Bioconductor dependencies separately first. (I've verified this on my own machine.) I've changed the recommended installation procedure to reflect this.
